### ISSUE TITLE
Option for delegate_to and local_action to run only once with a list

### ISSUE THREAD

**Issue by cheribral** [ansible/ansible#2389]
With delegate_to and local_action, ansible currently runs the command once for every host selected from inventory.  It would be nice to have a way to run the command only once with a list of the relevant hosts.  Running a batch job seems much more efficient than firing off a process hundreds of times, not to mention that the called program can be more intelligent and less cumbersome with knowledge of the entire list of hosts.

While looking around for how this could be done, I did find BYPASS_HOST_LOOP, but that requires a custom module poking around at inventory, good knowledge of ansible internals and generally just seems kind of nasty.

**Comment by bcoca** [comment]
one solution is 2 playbooks, the first runs the single actions on the local
host, the 2nd runs per host.

---

hosts: localhost
tasks:

##     - action: this is local

hosts: allothers
tasks:
- action: these are remote

## 

Brian Coca
Stultorum infinitus est numerus
0110000101110010011001010110111000100111011101000010000001111001011011110111010100100000011100110110110101100001011100100111010000100001
Pedo mellon a minno

**Comment by cheribral** [comment]
That would solve running it once, but it wouldn't solve the problem of knowing
which hosts to work on.  The commands are meant 'to perform a task on one host
with reference to other hosts'.  If we took the example in the documentation of
removing hosts from a load balancer, having it run in a play that only selects
localhost would not work very well.  Even if the new playbook could use magic
variables which populated the same way host selection does, that would require
one playbook to know what another selected, which is still dirty.  It would be
better to put it right in with the relevant playbook and host selection.

**Comment by bcoca** [comment]
ah, you want to batch and remove ALL hosts from load balancer in 1 action,
then operate on each host on the rest?

## I would still do the example above, just add with_items: to list the hosts

Brian Coca
Stultorum infinitus est numerus
0110000101110010011001010110111000100111011101000010000001111001011011110111010100100000011100110110110101100001011100100111010000100001
Pedo mellon a minno

**Comment by mpdehaan** [comment]
You don't even need two playbooks, you can do this with 2 plays in the same playbook.

This is a common misconception, a playbook is a list of plays, that's why it starts with "-".

Load balancer removal is best done with a delegate_to the load balancer and then you are good to go, because you are walking over your current host loop and talking to the LB for each host.

Perhaps I'm not understanding something but that seems already pretty well handled as a lot of folks are doing exactly this.

**Comment by cheribral** [comment]
The point wasn't about using two separate plays or playbooks, that was something mentioned by bcoca.

The load balancer removal example is indeed talking to the load balancer again and again for each and every host, which IS the point of what I brought up.  It would be nice not to launch that process over and over again.  If you know that you are going to be removing a group of servers, just send the list once and be done with it.

In any situation where it is cheaper to run a process all at once, it quickly becomes significant when you have a large number of hosts.  For instance, I don't want to trigger SNMP walks over parts of my network over and over again by repeatedly calling a process which does so on every invocation.  Sure, if you are just running "sed /$1/d someconfig" on every invocation, whatever, but to be generally useful for bigger tasks, it would be nice to run once with a list.

**Comment by paco0x** [comment]
This issue can be solved with [Run Once](http://docs.ansible.com/ansible/playbooks_delegation.html#run-once) option that is new in ansible 1.7
